File: app/src/main/java/dk/blekinge/dicerolla/CircularList.java
package dk.blekinge.dicerolla;

import java.util.ArrayList;
import java.util.List;

public class CircularList<Type> extends ArrayList<Type> {
    public Type get(int index) {
        if (this.isEmpty()) {
            throw new IndexOutOfBoundsException("List is empty");
        }
        return super.get(index % this.size());
    }
}

File: app/src/main/java/dk/blekinge/dicerolla/CircularList.kt
package dk.blekinge.dicerolla

import kotlin.collections.ArrayList

class CircularList<Type> : ArrayList<Type> {

    constructor(c: Collection<Type?>) : super(c)
    constructor() : super()

    override fun get(index: Int): Type {
        if (this.isEmpty()) {
            throw IndexOutOfBoundsException("List is empty")
        }
        return super.get(index % this.size)
    }

    override fun subList(fromIndex: Int, toIndex: Int): MutableList<Type> {
        return if (fromIndex >= size) {
            subList(fromIndex - size, toIndex - size)
        } else if (toIndex > size) {
            super.subList(fromIndex, size).plus(subList(0, toIndex - size)).toMutableList()
        } else {
            super.subList(fromIndex, toIndex)
        }
    }
}